tests/node_compat/test/parallel/test-v8-serdes.js

```diff
@@ -0,0 +1,284 @@
+// deno-fmt-ignore-file
+// deno-lint-ignore-file
+
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+// Taken from Node 18.12.1
+// This file is automatically generated by `tests/node_compat/runner/setup.ts`. Do not modify this file manually.
+
+// Flags: --expose-internals
+
+'use strict';
+
+const common = require('../common');
+const { internalBinding } = require('internal/test/binding');
+const assert = require('assert');
+const v8 = require('v8');
+const os = require('os');
+
+const circular = {};
+circular.circular = circular;
+
+const objects = [
+  { foo: 'bar' },
+  { bar: 'baz' },
+  new Int8Array([1, 2, 3, 4]),
+  new Uint8Array([1, 2, 3, 4]),
+  new Int16Array([1, 2, 3, 4]),
+  new Uint16Array([1, 2, 3, 4]),
+  new Int32Array([1, 2, 3, 4]),
+  new Uint32Array([1, 2, 3, 4]),
+  new Float32Array([1, 2, 3, 4]),
+  new Float64Array([1, 2, 3, 4]),
+  new DataView(new ArrayBuffer(42)),
+  Buffer.from([1, 2, 3, 4]),
+  new BigInt64Array([42n]),
+  new BigUint64Array([42n]),
+  undefined,
+  null,
+  42,
+  circular,
+];
+
+// TODO(nathanwhit): we don't have any exposed host objects, so these parts don't work
+// const hostObject = new (internalBinding('js_stream').JSStream)();
+
+{
+  const ser = new v8.DefaultSerializer();
+  ser.writeHeader();
+  for (const obj of objects) {
+    ser.writeValue(obj);
+  }
+
+  const des = new v8.DefaultDeserializer(ser.releaseBuffer());
+  des.readHeader();
+
+  for (const obj of objects) {
+    assert.deepStrictEqual(des.readValue(), obj);
+  }
+}
+
+{
+  for (const obj of objects) {
+    assert.deepStrictEqual(v8.deserialize(v8.serialize(obj)), obj);
+  }
+}
+
+{
+  const ser = new v8.DefaultSerializer();
+  ser._getDataCloneError = common.mustCall((message) => {
+    assert.strictEqual(message, '#<Object> could not be cloned.');
+    return new Error('foobar');
+  });
+
+  ser.writeHeader();
+
+  assert.throws(() => {
+    ser.writeValue(new Proxy({}, {}));
+  }, /foobar/);
+}
+
+// {
+//   const ser = new v8.DefaultSerializer();
+//   ser._writeHostObject = common.mustCall((object) => {
+//     assert.strictEqual(object, hostObject);
+//     const buf = Buffer.from('hostObjectTag');
+
+//     ser.writeUint32(buf.length);
+//     ser.writeRawBytes(buf);
+
+//     ser.writeUint64(1, 2);
+//     ser.writeDouble(-0.25);
+//   });
+
+//   ser.writeHeader();
+//   ser.writeValue({ val: hostObject });
+
+//   const des = new v8.DefaultDeserializer(ser.releaseBuffer());
+//   des._readHostObject = common.mustCall(() => {
+//     const length = des.readUint32();
+//     const buf = des.readRawBytes(length);
+
+//     assert.strictEqual(buf.toString(), 'hostObjectTag');
+
+//     assert.deepStrictEqual(des.readUint64(), [1, 2]);
+//     assert.strictEqual(des.readDouble(), -0.25);
+//     return hostObject;
+//   });
```

Comment:
Can you comment these tests why they are failing right now?

Reply:
Oh they all rely on `hostObject` / JSStream (see the comment on line 42). I can add comments to that effect